File: Data Structures & Algorithms/two-integer-sum/submission-0.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int, int> numToLoc;
        for(int i = 0; i < nums.size(); ++i){
            int diff = target - nums[i];
            if(numToLoc.find(diff) != numToLoc.end()){
                return {numToLoc[diff], i};
            }
            if(numToLoc.find(nums[i]) == numToLoc.end()){
                numToLoc[nums[i]] = i;
            }
        }
        return {-1, -1};
    }
};

File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        leftPtr, rightPtr = 0, len(numbers) - 1
        while leftPtr < rightPtr:
            if numbers[leftPtr] + numbers[rightPtr] == target:
                return [leftPtr+1, rightPtr+1]
            elif numbers[leftPtr] + numbers[rightPtr] > target:
                rightPtr-=1
            else:
                leftPtr+=1
        return [-1,-1]